rllib/agents/dqn/dqn.py

```diff
@@ -66,23 +90,23 @@
         "n_step": 1,
 
         # === Replay buffer ===
-        # Size of the replay buffer. Note that if async_updates is set, then
-        # each worker will have a replay buffer of this size.
+        # Deprecated, use capacity in replay_buffer_config instead.
         "buffer_size": DEPRECATED_VALUE,
-        # Prioritized replay is here since this algo uses the old replay
-        # buffer api
-        "prioritized_replay": True,
         "replay_buffer_config": {
             # For now we don't use the new ReplayBuffer API here
-            "_enable_replay_buffer_api": False,
-            "type": "MultiAgentReplayBuffer",
+            "_enable_replay_buffer_api": True,
+            "type": "MultiAgentPrioritizedReplayBuffer",
+            # Size of the replay buffer. Note that if async_updates is set,
+            # then each worker will have a replay buffer of this size.
             "capacity": 50000,
-            "replay_batch_size": 32,
             "prioritized_replay_alpha": 0.6,
             # Beta parameter for sampling from prioritized replay buffer.
             "prioritized_replay_beta": 0.4,
             # Epsilon to add to the TD errors when updating priorities.
             "prioritized_replay_eps": 1e-6,
+            # The number of continuous environment steps to replay at once. This may
+            # be set to greater than 1 to support recurrent models.
+            "replay_sequence_length": 1,
```

Comment:
Am I correct in that this is analogous to `max_sequence_length` from policy-gradient based policies? It might make sense to have a `full_episode` setting for variable-length episodes. The batch can be right zero-padded to the longest episode length in the train batch.

Reply:
Yes, the replayed sequences is often shorter than `replay_sequence_length`.
What you are describing can be accomplished by setting `storage_unit="episodes"` or `storage_unit=StorageUnit.EPISODES`. Padding of the batch is so far not handled by the buffers, which is open for discussion! I think it should not be handled by buffers, especially so that buffers can be reinstantiated from checkpoints in any setting.